### DISCUSSION TITLE
How to make stitching faster?

### DISCUSSION THREAD

**Discussion by fdelu** [OpenStitching/stitching#256] [Q&A]
Hello!

I am currently using the Stitcher to stitch 15 OpenCV images of a fixed-position camera that pans from left to right. All the images are 1920x1080 images captured with OpenCV's VideoCapture. I am looking for advice/guidance on how to speed up the stitching. Is there anything I can look into to speed it up?

Some additional information:

- I am using the Stitcher with the following parameters: `Stitcher(confidence_threshold=0, range_width=1)` (since all images are guaranteed to be ordered left to right)
- I've tried to resize the images to make them 25% smaller, doesn't seem to have helped a lot
- All the images are in memory, I am not saving anything to disk nor encoding any of the images but the final panorama
- Stitching is taking about 2 minutes for the 15 images. The CPU I am using is not very fast unfortunately.

Is there something else I can try?

**Comment by lukasalexanderweber** [comment]
two things that come in my mind:

1. You could safe the homography that it's not reestimated every time: https://github.com/OpenStitching/stitching/discussions/148
2. You could reduze the final image size (`final_megapix`)
